Firestore/Source/Public/FirebaseFirestore/FIRLoadBundleTask.h

```diff
@@ -28,28 +34,57 @@ typedef NS_ENUM(NSInteger, FIRLoadBundleTaskState) {
 
 } NS_SWIFT_NAME(LoadBundleTaskState);
 
+/** Represents a progress update or a final state from loading bundles. */
 NS_SWIFT_NAME(LoadBundleTaskProgress)
 @interface FIRLoadBundleTaskProgress : NSObject
 
+/** How many documents have been loaded. */
 @property(readonly, nonatomic) NSInteger documentsLoaded;
+
+/** The total number of documents in the bundle. 0 if the bundle failed to parse. */
 @property(readonly, nonatomic) NSInteger totalDocuments;
+
+/** How many bytes have been loaded. */
 @property(readonly, nonatomic) NSInteger bytesLoaded;
+
+/** The total number of bytes in the bundle. 0 if the bundle failed to parse. */
 @property(readonly, nonatomic) NSInteger totalBytes;
 
+/** The current state of `FIRLoadBundleTask` (`LoadBundleTask` in Swift). */
 @property(readonly, nonatomic) FIRLoadBundleTaskState state;
 
 @end
 
+/** A handle associated with registered observers, that can be used to remove them. */
 typedef NSInteger FIRLoadBundleHandle NS_SWIFT_NAME(LoadBundleHandle);
 
+/**
+ * Represents the task of loading a Firestore bundle. Observers can be registered with this task to
+ * observe progresses of bundle loading, as well as task completion and error events.
+ */
 NS_SWIFT_NAME(LoadBundleTask)
 @interface FIRLoadBundleTask : NSObject
 
-- (FIRLoadBundleHandle)observeWithHandler:(void (^)(FIRLoadBundleTaskProgress *progress))handler;
+/**
+ * Registers an observer to observe the progress updates, completion or error events.
+ *
+ * @return A handle to the registered observer, which can be used to remove the observer once it is
+ * no longer needed.
+ */
+- (FIRLoadBundleHandle)observeWith:(void (^)(FIRLoadBundleTaskProgress *progress))handler
```

Comment:
Hmm, I'm a little surprised `Handler` is removed from the method name. I'm not very familiar with the Objective-C naming conventions, but I thought the parameter name is generally repeated in the function name. What is the reason for this change?

Reply:
It's mainly because I found it very awkward to have both `handler` in code and `observer` in comment referring to pretty much the same thing.

I am open to suggestions.

Reply:
What do you think about `addObserver` instead? Also I think `FIRLoadBundleHandle` should be `FIRLoadBundleObserverHandle`.

Reply:
Agree on both (you probably need to talk to the API council, but IIUC, you're already on that)